### ISSUE TITLE
🐛 [Firebase Firestore] swift Force Load _swiftCompatabilityConcurancy

### ISSUE THREAD

**Issue by waleedkalyar** [firebase/flutterfire#9464]
Hi sir, I have implemented flutter fire inside my flutter project. For reducing time I have added build in C++ libs in Pod file as flutter fire documentation says. But whenever I try to build its give me the error of

Undefined symbol: __swift_FORCE_LOAD_$_swiftCompatabilityConcurancy

I am unable to figure out the problem.
Btw it will build if I remove 
pod 'FirebaseFirestore', :git => 'https://github.com/invertase/firestore-ios-sdk-frameworks.git', :tag => '9.4.0'
from my Pod file. But It will take almost one hour on my MacBook to build.

My OS is -> Mac OS big sure version 11.5.2
My Xcode version is -> 13.0.0
Its become pain in my head to use firebase Firestore. If I remove Firestore dependency build work like charm. 
Please help me to sort this problem.

**Comment by darshankawar** [comment]
@waleedkalyar 
Can you provide `flutter doctor -v`, complete error log along with `pubspec.yaml` and `ios/podfile` in properly formatted text.

**Comment by waleedkalyar** [comment]
### Flutter Doctor Summery
aliata@Alis-MacBook-Pro onesite % flutter doctor -v
[✓] Flutter (Channel stable, 3.0.5, on macOS 11.5.2 20G95 darwin-x64, locale en-PK)
    • Flutter version 3.0.5 at /Volumes/Stuff/flutterSDK/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (8 weeks ago), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2

[!] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /Users/aliata/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.

[!] Xcode - develop for iOS and macOS (Xcode 13.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! CocoaPods 1.10.0 out of date (1.11.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] VS Code (version 1.70.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (3 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86    • Android 10 (API 29) (emulator)
    • macOS (desktop)                    • macos         • darwin-x64     • macOS 11.5.2 20G95 darwin-x64
    • Chrome (web)                       • chrome        • web-javascript • Google Chrome 104.0.5112.101

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 2 categories.

### Pods File
`# Uncomment this line to define a global platform for your project
platform :ios, '10.0'

ENV['COCOAPODS_DISABLE_STATS'] = 'true'

project 'Runner', {
  'Debug' => :debug,
  'Profile' => :release,
  'Release' => :release,
}

def flutter_root
  generated_xcode_build_settings_path = File.expand_path(File.join('..', 'Flutter', 'Generated.xcconfig'), __FILE__)
  unless File.exist?(generated_xcode_build_settings_path)
    raise "#{generated_xcode_build_settings_path} must exist. If you're running pod install manually, make sure flutter pub get is executed first"
  end

  File.foreach(generated_xcode_build_settings_path) do |line|
    matches = line.match(/FLUTTER_ROOT\=(.*)/)
    return matches[1].strip if matches
  end
  raise "FLUTTER_ROOT not found in #{generated_xcode_build_settings_path}. Try deleting Generated.xcconfig, then run flutter pub get"
end

require File.expand_path(File.join('packages', 'flutter_tools', 'bin', 'podhelper'), flutter_root)

flutter_ios_podfile_setup

target 'Runner' do
  use_frameworks!
  use_modular_headers!
  pod 'FirebaseFirestore', :git => 'https://github.com/invertase/firestore-ios-sdk-frameworks.git', :tag => '9.4.0'
  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
  end
end
`

**Comment by darshankawar** [comment]
@waleedkalyar 
If you change to `9.3.0` and try, do you get same error ?
Also, can you try to delete `podfile.lock`, run `flutter clean` followed by `pod repo update` and see if it helps ?

**Comment by waleedkalyar** [comment]
I have tried with both  `9.3.0` and `9.0.0`   delete   `pod file.lock`   And already done all these steps.
But still the issue is:
`Error (Xcode): Undefined symbol: __swift_FORCE_LOAD_$_swiftCompatibilityConcurrency`

**Comment by waleedkalyar** [comment]
### The Full log of system is @darshankawar 
`  Undefined symbols for architecture x86_64:
      "__swift_FORCE_LOAD_$_swiftCompatibilityConcurrency", referenced from:
          __swift_FORCE_LOAD_$_swiftCompatibilityConcurrency_$_FirebaseFirestoreSwift in FirebaseFirestoreSwift(CodableErrors.o)
          __swift_FORCE_LOAD_$_swiftCompatibilityConcurrency_$_FirebaseFirestoreSwift in FirebaseFirestoreSwift(CollectionReference+AsyncAwait.o)
          __swift_FORCE_LOAD_$_swiftCompatibilityConcurrency_$_FirebaseFirestoreSwift in FirebaseFirestoreSwift(DocumentID.o)
          __swift_FORCE_LOAD_$_swiftCompatibilityConcurrency_$_FirebaseFirestoreSwift in FirebaseFirestoreSwift(DocumentReference+Codable.o)
          __swift_FORCE_LOAD_$_swiftCompatibilityConcurrency_$_FirebaseFirestoreSwift in FirebaseFirestoreSwift(DocumentReference+ReadDecodable.o)
          __swift_FORCE_LOAD_$_swiftCompatibilityConcurrency_$_FirebaseFirestoreSwift in FirebaseFirestoreSwift(DocumentSnapshot+ReadDecodable.o)
          __swift_FORCE_LOAD_$_swiftCompatibilityConcurrency_$_FirebaseFirestoreSwift in FirebaseFirestoreSwift(ExplicitNull.o)
          ...
         (maybe you meant: __swift_FORCE_LOAD_$_swiftCompatibilityConcurrency_$_FirebaseFirestoreSwift)
    ld: symbol(s) not found for architecture x86_64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    note: Using new build system
    note: Planning
    note: Build preparation complete
    note: Building targets in parallel

    Result bundle written to path:
        /var/folders/p5/8yd8xy253g78z85733whxz900000gn/T/flutter_tools.ucU7mc/flutter_ios_build_temp_dirtj2olx/temporary_xcresult_bundle


Error (Xcode): Undefined symbol: __swift_FORCE_LOAD_$_swiftCompatibilityConcurrency


Could not build the application for the simulator.
Error launching application on iPhone 11.
`

**Comment by darshankawar** [comment]
> `  Undefined symbols for architecture x86_64:
>       "__swift_FORCE_LOAD_$_swiftCompatibilityConcurrency", referenced from:
>           __swift_FORCE_LOAD_$_swiftCompatibilityConcurrency_$_FirebaseFirestoreSwift in FirebaseFirestoreSwift(CodableErrors.o)
>           __swift_FORCE_LOAD_$_swiftCompatibilityConcurrency_$_FirebaseFirestoreSwift in FirebaseFirestoreSwift(CollectionReference+AsyncAwait.o)
>           __swift_FORCE_LOAD_$_swiftCompatibilityConcurrency_$_FirebaseFirestoreSwift in FirebaseFirestoreSwift(DocumentID.o)
>           __swift_FORCE_LOAD_$_swiftCompatibilityConcurrency_$_FirebaseFirestoreSwift in FirebaseFirestoreSwift(DocumentReference+Codable.o)
>           __swift_FORCE_LOAD_$_swiftCompatibilityConcurrency_$_FirebaseFirestoreSwift in FirebaseFirestoreSwift(DocumentReference+ReadDecodable.o)
>           __swift_FORCE_LOAD_$_swiftCompatibilityConcurrency_$_FirebaseFirestoreSwift in FirebaseFirestoreSwift(DocumentSnapshot+ReadDecodable.o)
>           __swift_FORCE_LOAD_$_swiftCompatibilityConcurrency_$_FirebaseFirestoreSwift in FirebaseFirestoreSwift(ExplicitNull.o)
>           ...
>          (maybe you meant: __swift_FORCE_LOAD_$_swiftCompatibilityConcurrency_$_FirebaseFirestoreSwift)
>     ld: symbol(s) not found for architecture x86_64

This is probably pointing to native iOS Firebase sdk, so I suggest you to open this issue in dedicated repo: https://github.com/firebase/firebase-ios-sdk/issues

Also, check this and see if it helps.https://github.com/facebookarchive/react-native-fbsdk/issues/794#issuecomment-695217977

https://github.com/facebookarchive/react-native-fbsdk/issues/794#issuecomment-702481313

Closing from here, as this doesn't seem to be flutterfire issue. If you get directed here, write in comments and I'll reopen it.
